samples/templates/default-azuredeploy.json

```diff
@@ -340,6 +352,34 @@
                 "[resourceId('Microsoft.KeyVault/vaults', variables('serviceName'))]",
                 "[resourceId('Microsoft.DocumentDb/databaseAccounts', variables('serviceName'))]"
             ]
+        },
+        {
+            "name": "[parameters('storageAccountName')]",
+            "type": "Microsoft.Storage/storageAccounts",
+            "apiVersion": "2019-06-01",
+            "location": "[resourceGroup().location]",
+            "properties": {
+                "supportsHttpsTrafficOnly": "true"
+            },
+            "dependsOn": [],
+            "sku": {
+                "name": "Standard_LRS"
+            },
+            "kind": "Storage",
+            "tags": {}
+        },
+        {
+            "type": "Microsoft.Storage/storageAccounts/providers/roleAssignments",
+            "apiVersion": "2018-09-01-preview",
+            "name": "[concat(parameters('storageAccountName'), '/Microsoft.Authorization/', guid(uniqueString(parameters('storageAccountName'))))]",
+            "dependsOn": [
+                "[parameters('storageAccountName')]",
+                "[variables('serviceName')]"
+            ],
+            "properties": {
+                "roleDefinitionId": "[variables('roleDefinitionId')]",
+                "principalId": "[reference(concat('Microsoft.Web/sites/', variables('serviceName')), '2018-11-01', 'full').identity.principalId]"
```

Comment:
Should minimize to only permission needed (e.g., blob)

Reply:
The `roleDefinitionId` variable is defined to be the guid that corresponds to the `Storage Blob Data Contributor` role (Reference - https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles). If you are referring to the 'full' in the `reference` call, that just corresponds to returning the full object (so that we can get the `identity.principalId` of the app service.